ospatch/patch_run.go

```diff
@@ -415,7 +422,15 @@ func (r *patchRun) reportPatchDetails(patchState osconfigpb.Instance_PatchState,
 		res, err := r.client.ReportPatchJobInstanceDetails(r.ctx, &request)
 		if err != nil {
 			if s, ok := status.FromError(err); ok {
-				return fmt.Errorf("code: %q, message: %q, details: %q", s.Code(), s.Message(), s.Details())
+				err := fmt.Errorf("code: %q, message: %q, details: %q", s.Code(), s.Message(), s.Details())
+				switch s.Code() {
+				// Errors we should retry.
+				case codes.DeadlineExceeded, codes.Unavailable, codes.Aborted, codes.Internal:
```

Comment:
We should also retry ResourceExhausted errors, ideally with an extended, jittered wait interval.